File: app/src/main/java/com/sg/moviesindex/db/typeconverters/ReviewTypeConverter.java
package com.sg.moviesindex.db.typeconverters;

import androidx.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.sg.moviesindex.model.tmdb.Review;

import java.lang.reflect.Type;
import java.util.ArrayList;

public class ReviewTypeConverter {

    private static final Gson gson = new Gson();

    @TypeConverter
    public static ArrayList<Review> gettingListFromString(String data) {
        if (data == null) {
            return new ArrayList<Review>();
        }

        Type listType = new TypeToken<ArrayList<Review>>() {
        }.getType();

        return gson.fromJson(data, listType);
    }

    @TypeConverter
    public static String gettingStringFromList(ArrayList<Review> someObjects) {
        return gson.toJson(someObjects);
    }

}

File: app/src/main/java/com/sg/moviesindex/data/local/typeconverters/ReviewTypeConverter.kt
package com.sg.moviesindex.data.local.typeconverters

import androidx.room.TypeConverter
import com.sg.moviesindex.data.remote.Review
import com.squareup.moshi.Moshi
import com.squareup.moshi.Types

class ReviewTypeConverter {
  private val moshi = Moshi.Builder().build()
  private val listType = Types.newParameterizedType(List::class.java, Review::class.java)
  private val adapter = moshi.adapter<List<Review>>(listType)

  @TypeConverter
  fun gettingListFromString(data: String?): List<Review> {
    if (data == null) {
      return emptyList()
    }
    return adapter.fromJson(data) ?: emptyList()
  }

  @TypeConverter
  fun gettingStringFromList(someObjects: List<Review>): String = adapter.toJson(someObjects)
}
